src/main/content/engine/engine-content.ts

```
@@ -84,7 +85,9 @@ export class EngineContentAPI extends AbstractContentAPI<string, EngineVersion>
             });
     }
 
-    public async downloadEngine(engineVersion: string) {
+    public downloadEngine: DownloadEngine = async (engineVersion) => {
+        if (!engineVersion) return;
```

You should either throw an Error or assert that engineVersion is defined and not just return silently. It's important because we want it to crash so we know something is not right.

That's true every time you did this in this PR.